### ISSUE TITLE
report [ ]-loop resorting to serial iteration

### ISSUE THREAD

**Issue by vasslitvinov** [chapel-lang/chapel#12788]
The user may want to know when a [ ]-loop in their code resorts to serial computation. In some cases it is OK for the loop to run serially, as long as the user knows about it.

This issue requests generating such an error. It should include source file/line (as usual) as well as an indication of the reason why it resorted to serial.

**Comment by BryantLam** [comment]
As an idea, it would make sense to me to have a compiler flag like `--parallelization-report` or `--performance-report` generate a report akin to a vectorization report in C/C++ compilers. It could accept optional "level" integers to only show "executed as serial" diagnostic messages as in this issue here, or higher levels to be more chatty about where all [ ]-loops are parallel or serial, distributions used on arrays, etc.

Or instead of level integers, it could accept an item list to report what the user explicitly wants from a compile-time performance-debugging standpoint.

E.g., `chpl --performance-report=executed_as_serial,distribution_maps(nl=4),vectorized`

**Comment by bradcray** [comment]
I agree that this should be opted into via a flag of some sort (the original issue didn't make that clear, though I imagine that's what Vass was imagining?)

**Comment by vasslitvinov** [comment]
> this should be opted into via a flag of some sort

Indeed. I updated the issue indicating this.
